drools/drools-core-static/src/main/java/org/drools/statics/StaticKieAssemblers.java

```diff
@@ -40,6 +40,10 @@ public StaticKieAssemblers() {
         //         instance("org.kie.dmn.core.assembler.DMNAssemblerService"));
     }
 
+    @Override
+    public void addResourceBeforeRules( Object o, Resource resource, ResourceType resourceType, ResourceConfiguration resourceConfiguration ) throws Exception {
```

Comment:
I think this should work 

```suggestion
    public void addResourceBeforeRules(Object knowledgeBuilder, Resource resource, ResourceType type, ResourceConfiguration configuration) throws Exception {
        KieAssemblerService assembler = assemblers.get(type);
        if (assembler != null) {
            assembler.addResourceBeforeRules(knowledgeBuilder,
                                             resource,
                                             type,
                                             configuration);
        } else {
            throw new RuntimeException("Unknown resource type: " + type);
        }
    
```